### ISSUE TITLE
Runs entire suite on rerun instead of just failed tests in Azure DevOps pipeline

### ISSUE THREAD

**Issue by dotkarl** [joaoopereira/dotnet-test-rerun#39]
First of all, thank you for creating this tool! I recently started using it to rerun our integration tests in our Azure DevOps pipeline. This suite suffers from some ordering effects. We are trying to resolve that issue, but in the meantime we would like to keep our build green. _dotnet-test-rerun_ seems to be made for that use case!

However, I noticed something does not seem to work as intended -- at least in our pipeline. After each test, _test-rerun_ runs the entire test suite again, not just the failed tests. (That is the intended behaviour, right?) 

This is the output I see in the integration test task:

```
Rerun attempt 3/3
Found Failed tests. Rerun filter: [...failing tests...]
working directory: /
forking test [...path...]/[...application...].IntegrationTests.dll --logger "trx" --results-directory [...directory...]
```

Now I suspect that I should also see `--filter` in that last line of code -- is that correct? I've been investigating the code base to confirm my suspicion, but I haven't found the place where the filter is set just yet.

This is the script we execute in our build task:

```
dotnet tool install --global dotnet-test-rerun
test-rerun [...path...]/[...application...].IntegrationTests.dll --results-directory ../../_temp
```

We need to specify the results directory because the task did not have write access at the original location.

So, in short, two questions:
1) How does _dotnet-test-rerun_  decide which tests to rerun? Where in the code can I find this logic?
2) Do you have any idea why it reruns our entire integration test suite, instead of just the failing tests?

**Comment by joaoopereira** [comment]
HI @dotkarl,

many thanks for using our tool :)

We will take a look on this ASAP (we hope today)

**Comment by joaoopereira** [comment]
@dotkarl issue fixed in version 1.2.1.

thanks @ricardofslp 💪
